src/main/ts/api.ts

```diff
@@ -73,3 +113,14 @@ export function getIssues() {
     return chartData;
   });
 }
+
+export function getProjects({ q }: { q: string }) {
+  return getJSON('/api/components/search_projects', q ? { q } : undefined)
```

Comment:
The filtering mechanism is not working this way. Please use: filter='query = "Sonar"'